src/utils/helpers/completionDateTimes.js

```diff
@@ -1,14 +1,44 @@
 import { priorityCodeCompletionTimes } from '../hact/helpers/priority-codes'
 import add from 'date-fns/add'
 
-const WEEKEND_DAYS = [0, 6]
+const SATURDAY_DAY_NUMBER = 6
+const SUNDAY_DAY_NUMBER = 0
 
-const isWorkingDay = (date) => !WEEKEND_DAYS.includes(date.getDay())
+const WEEKEND_DAY_NUMBERS = [SATURDAY_DAY_NUMBER, SUNDAY_DAY_NUMBER]
+
+const isWorkingDay = (date) => !WEEKEND_DAY_NUMBERS.includes(date.getDay())
+
+// This function exists so we can "offset" the starting count of working days
+// for an order which is raised on a weekend. This is a special case where
+// the number of working days used in the target date calculation actually should
+// include the days over the weekend in which the order was raised.
+
+// For example, an urgent order raised on a Saturday should have a target
+// date of the following *Thursday* even though such orders have a target of 5
+// working days. This function provides an offset in those cases so
+// the weekend "counts" towards the working days count.
+const startingDayCountOffset = (date, targetCount) => {
+  if (date.getDay() === SATURDAY_DAY_NUMBER && targetCount > 2) {
+    return 2
+  } else if (date.getDay() === SUNDAY_DAY_NUMBER && targetCount > 1) {
+    return 1
+  } else {
+    return 0
+  }
+}
 
 const dateAfterCountWorkingDays = (startDate, targetWorkingDaysCount) => {
-  var workingDaysCount = 0
-  var calendarDaysCount = 0
+  // If raised on a weekend, include the remainder of that weekend for
+  // the count used to create the returned date
+  var startingOffsetDayCount = startingDayCountOffset(
```

Comment:
Think we could probably use ES6 `let` instead of `var` when variable re-assignment may happen.